### PR TITLE
MATAD02 - Product-Artist association

### PR DESCRIPTION
**Pull request by ivancosmin**
### What does this PR change? 
Product - Artist association

Identify artist metaobject and associate it with the product

### CODE REVIEW

File: app/Console/Commands/ProductsArtistsSync.php [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+        $query
+            ->lazyById()
+            ->each(function (ShopifyProduct $product) use ($progressBar) {
+                $migrationOriginData = $product->migrationProduct()->value('origin_data');
+
+                foreach($migrationOriginData['custom_attributes'] as $attribute) {
+                    if ('artist' === $attribute['attribute_code']) {
+                        $artistSlug = preg_replace('/\W+/', '-', $attribute['value']);
+
+                        $artist = $this->shopifyAdapter->getArtistMetaobjectBySlug($artistSlug);
```
Comment (radmen) [inline app/Console/Commands/ProductsArtistsSync.php:53]: :large_blue_circle: I strongly recommend wrapping this one using `once()`:


```suggestion
                        $artist = once(function () use ($artistSlug) {
                            return $this->shopifyAdapter->getArtistMetaobjectBySlug($artistSlug);
                        });
```

That way, this part of code would be executed only once, and the results will be kept in memory.

For older versions of Laravel, you would have to put those in array cache, like this:

```php
cache()
    ->store('array')
    ->rememberForever('cache key', fn () => $this->shopify->getArtistMetaobjectBySlug('...'));
```

File: app/Console/Commands/ProductsArtistsSync.php [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+                        if (!is_null($artist)) {
+                            $this->shopifyAdapter->setMetafields([
+                                [
+                                    'key' => 'artist',
+                                    'namespace' => 'beggars',
+                                    'value' => $artist->id,
+                                    'type' => 'single_line_text_field',
```
Comment (radmen) [inline app/Console/Commands/ProductsArtistsSync.php:61]: :red_circle: This is not a correct type. It should be `metaobject_reference`

**Review by radmen — CHANGES_REQUESTED** [review]
I have finished reviewing your Pull Request. I have flagged my comments with the following emojis:

- 🔴 indicates that changes are required
- 🔵 suggests making a change
- 💬 is a tip, general feedback, or a comment
- 🤔 is a question
  
Please review my comments and make the necessary changes. If I left any code suggestions, they might not work. Check them first before making any commits. 

If you have any questions or concerns, please let me know.

File: app/Console/Commands/ProductsArtistsSync.php (hunk 1 above)
Comment (ivancosmin) [reply app/Console/Commands/ProductsArtistsSync.php]: pushed the changes

**Review by radmen — APPROVED** [review]

**Pull request merged**